deltametrics/cube.py

```diff
@@ -1,21 +1,406 @@
+    number of attached attributes (grain size, mud frac, elevation).
 
     .. note::
-        This object is going to be core to the success of the project.
-    
+        `Cube` does not load any data into memory by default. This means that
+        slicing is handled "behind the scenes" by an :doc:`I/O file handler
+        </reference/io/index>`. Optionally, you can load files into memory for (sometimes) faster operations.
+        See the :meth:`read` for more information.
+
     """
-    def __init__(self, *args):
+
+    def __init__(self, data, read=[], varset=None):
         """Initialize the Cube.
+
+        Parameters
+        ----------
+        data : :obj:`str`, :obj:`dict`
+            If data is type `str`, the string points to a NetCDF or HDF5 file
+            that can be read. Typically this is used to directly import files
+            output from the pyDeltaRCM model. Alternatively, pass a
+            :obj:`dict` with keys indicating variable names, and values with
+            corresponding t-x-y `ndarray` of data.
+
+        read : :obj:`bool`, optional
+            Which variables to read from dataset into memory. Special option
+            for ``read=True`` to read all available variables into memory.
+
+        varset : :class:`~deltametrics.plot.VariableSet`, optional
+            Pass a valid `~deltametrics.plot.VariableSet` instance if you wish
+            to style this cube similarly to another cube.
         """
 
-        pass
+        # self.data = data
+        # self.data = _DataDict({})
+        if type(data) is str:
+            self._data_path = data
+            self._connect_to_file(data_path=data)
+            self._read_meta_from_file()
+            # self._connect_vars_to_file(self.variables)
+            self._compute_strata()
+        elif type(data) is dict:
+            # handle a dict, arrays set up already, make an io class to wrap it
+            self._data_path = None
+            raise NotImplementedError
+        else:
+            raise TypeError('Invalid type for "data": %s' % type(data))
+
+        self._plan_set = {}
+        self._section_set = {}
+
+        if varset:
+            self.varset = varset
+        else:
+            self.varset = plot.VariableSet()
+
+    def __getitem__(self, var):
+        """Return the variable.
+
+        Overload slicing operations for io to return a
+        :obj:`~deltametrics.cube.CubeVariable` instance when slicing.
+
+        Parameters
+        ----------
+        var : :obj:`str`
+            Which variable to slice.
+
+        Returns
+        -------
+        CubeVariable : `~deltametrics.cube.CubeVariable`
+            The instantiated CubeVariable.
+        """
+
+        if var in self._variables:
+            return CubeVariable(self.dataio[var], variable=var)
+        elif var == 'time':
+            # a special attribute we add, which matches eta.shape
+            _eta = self.dataio[var]
```

Comment:
`_eta` is just a view of the `eta` variable, created temporarily, to access it's shape attribute. I guess we could get the array shape during `Cube` instantiation, and then store it. That may be better.

Ultimately though, this is likely to have to change anyway when we create a new standard for the file structures.

Yeah of course, anything can vary with time. Right now, I have things set up to assume that all variables have a common "time" vector. As we've discussed, we may want to have time be a vector defined for each variable individually. If that's the case, we would create the time

> I see eta elsewhere as a special case, rather than var.

I guess this is probably because `eta` dictates what gets preserved in stratigraphy, whereas `sand_frac`, `velocity`, etc, do not. So `eta` is kind of special in that way.